tests/imageuploadengine.js

```diff
@@ -234,4 +235,48 @@ describe( 'ImageUploadEngine', () => {
 		expect( loader.status ).to.equal( 'aborted' );
 		sinon.assert.calledOnce( abortSpy );
 	} );
+
+	it( 'image should be permanently removed if it is removed by user during upload', done => {
```

Comment:
Honestly, I hate this test and how you need to use three consecutive `document#event:changesDone` to test something "just after the image placeholder is removed".

It's hard to test it because the image is removed in a `Promise`'s catch callback and the `Promise` is not returned anywhere. This test (and a test above that was already there) are very prone to fail in future because of theoretically unrelated changes.